File: 0058.最后一个单词的长度/0058-最后一个单词的长度.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        # s = s.split(' ')
        # re_list = []
        # for i in s:
        #     if i != '':
        #         re_list.append(i)
        # return len(re_list[-1])
        n = len(s)
        ans = 0
        for i in range(n-1, -1, -1):
            if s[i] != ' ':
                ans += 1
            else:
                if ans:
                    return ans
        return ans

File: 0058.最后一个单词的长度/0058-最后一个单词的长度.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        // int flag = 0;
        // int result = 0;
        // for( int i = s.size()-1; i >=0 ; i--){
        //     if( flag == 1 && s[i] == ' '){
        //         return result;
        //     }
        //     if( s[i] != ' ' ){
        //         flag = 1;
        //         result++;
        //     }
        // }
        // return result;
        int index = s.size()-1;
        while( s[index] == ' '){
            index--;
        }
        int result = 0;
        while(index >= 0 && s[index] != ' '){
            result++;
            index--;
        }
        return result;

    }
};